runtime/freelist.c

```diff
@@ -188,6 +189,7 @@ static header_t *nf_allocate (mlsize_t wo_sz)
     ++instr_size[19];
   }
 #endif /* CAML_INSTR */
+    caml_ev_alloc(wo_sz);
```

Comment:
The code here for tracking allocations is inside the `nf_allocate` functions, which is only one of the three allocators that OCaml now supports. As well as next-fit, there's also `ff_alllocate` (first-fit) and `bf_allocate` (best-fit).

Rather than duplicating this logic per-allocator, a better place for it is the `caml_alloc_shr_aux` function in `memory.c`, which is the common entry point that dispatches to the current allocator.